src/src/PluginStructs/P145_data_struct.cpp

```diff
@@ -312,16 +415,27 @@ float P145_data_struct::getCorrectedPPM(float rSensor, float temperature, float
       {
         c = sensordef.core * temperature + sensordef.corf * humidity + sensordef.corg;
       }
-    case p145AlgB:
-      // TODO Still missing a formula to correct for temp/hum when applying this algorithm
       break;
+    case p145AlgB:
     case p145AlgC:
-      // TODO Still missing a formula to correct for temp/hum when applying this algorithm
+      // Assume a 2nd order polynomial for temperature and 1st order poynomial for humidit
+      // A + B*t + C*t*t + D*h
+      c = (((sensordef.corc * temperature ) + sensordef.corb) * temperature) + sensordef.cora + (sensordef.cord * humidity);
       break;
-    default:
-      // Do nothing, return a default value at the end
+    case p145AlgNone:
+      // no correction
       break;
+    default:
+      // No correction
+      c = 1.0f;
+  }
+  if (c <=0)  // In case no or wrong correction factors are spefified disable correction (multiplier = 1.0)
```

Comment:
Float compare should also apply the decimal and f postfix, `0.0f`.
There's a typo in 'spefified'...